src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -51,17 +51,11 @@ public IConfigurationProvider Build(IConfigurationBuilder builder)
             }
             catch (InvalidOperationException ex) // InvalidOperationException is thrown when any problems are found while configuring AzureAppConfigurationOptions or when SDK fails to create a configurationClient.
             {
-                if (!_optional)
-                {
-                    throw new ArgumentException(ex.Message, ex);
-                }
+                throw new ArgumentException(ex.Message, ex);
             }
             catch (FormatException fe) // FormatException is thrown when the connection string is not a well formed connection string.
             {
-                if (!_optional)
-                {
-                    throw new ArgumentException(fe.Message, fe);
-                }
+                throw new ArgumentException(fe.Message, fe);
```

Comment:
Could you also update the summary of optional flag in `AzureAppConfigurationExtensions`?
Maybe we can change the first sentence to something like - "Determines the behavior of the App Configuration provider when an exception occurs while loading data from server."

Reply:
Good catch!